_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp

```diff
@@ -3854,8 +3855,11 @@ mfxStatus MfxHwH264Encode::CheckVideoParamQueryLike(
         changed = true;
     }
 
-    if (par.mfx.RateControlMethod == MFX_RATECONTROL_ICQ ||
-        par.mfx.RateControlMethod == MFX_RATECONTROL_LA_ICQ)
+    if (par.mfx.RateControlMethod == MFX_RATECONTROL_ICQ)
+    {
+        if (!CheckRange(par.mfx.ICQQuality, 1, 51)) changed = true;
+    }
+    if (par.mfx.RateControlMethod == MFX_RATECONTROL_LA_ICQ)
```

Comment:
Why we need such a change?

Reply:
In VAAPI, the comment for ICQ_quality_factor is that "Initial quality factor used in ICQ mode." and "This value must be between 1 and 51.". So I think in MSDK, the value check for it should align with the definition in VAAPI. @lizhong1008

Reply:
If MSDK still checks the value in the range of [0, 51], and doesn't have process for 0, then 0 can be transferred to media driver. At present, the valid range for ICQ quality factor in media driver is [1, 51], if 0 is set, then media driver will report error. Currently, I haven't seen there is process for 0. If I missed this process, please correct me. Thanks! @dvrogozh

Reply:
@ogolikov, @takulako, please look.